Create ui-c8y-1021-56-1-make-session-state-optional-in-sso-login-flow-8555-graft-release.md (#3139)
SoftwareAG/c8y-docs: content/change-logs/application-enablement/ui-c8y-1021-56-1-make-session-state-optional-in-sso-login-flow-8555-graft-release.md

---
date: ""

> title: Make session_state optional in SSO login flow

Reviewer: ```suggestion
title: Changed session_state property to be optional in the SSO login flow
```

product_area: Application enablement & solutions
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:

>   - value: component-YbYJ3gLU_

Reviewer: ```suggestion
  - value: q3kclF6pO
```

>     label: Web SDK

Reviewer: ```suggestion
    label: Authentication
```

build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-62824
version: 1021.56.1
---

> Make session_state optional in SSO login flow. (#8555) [GRAFT][release/cd] (#8744)

Reviewer 2: ```suggestion
When 'Redirect to the user interface application' is enabled, the UI application was requiring 'session_state' and 'code' properties in response to authorization request from the OAuth server. There are cases where 'session_state' is not returned by oauth server but 'code' only and login was failing. Session state is optional now.
```

> When 'Redirect to the user interface application' is enabled, the UI application was requiring 'session_state' and 'code' properties in response to authorization request from the OAuth server. There are cases where 'session_state' is not returned by oauth server but 'code' only and login was failing. Session state is optional now.

Reviewer: ```suggestion
Previously, when the option **Redirect to the user interface application** was enabled in the single sign-on login configuration, the UI application required the `session_state` and `code` properties in the response to the authorization request from the OAuth server. In certain cases the `session_state` property is not returned by the OAuth server which caused a login failure. Now the `session_state` property is optional and the login no longer fails due to this property missing.
```
